h2/src/main/org/h2/mvstore/db/MVTableEngine.java

```diff
@@ -162,7 +162,7 @@ void open(Database db, MVStore.Builder builder, boolean encrypted) {
                 }
                 this.transactionStore = new TransactionStore(
                         store,
-                        new ValueDataType(null, db, null));
+                        new ValueDataType(db.getCompareMode(), db, null));
```

Comment:
I understand the other change, but why this one?

Reply:
Because after you make the first change, it will give NPE on compareMode later on (I do not have stackt race handy). This one is a correct one actually, unlike the first, which is a total hack 8-)

Reply:
It is in ValueString.compareSecure(ValueString.java:47)